### ISSUE TITLE
L016: Comment clauses being reported to be long

### ISSUE THREAD

**Issue by nitinprakash96** [sqlfluff/sqlfluff#3480]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Take the following SQL for example:
```sql
CREATE TABLE IF NOT EXISTS foo
( id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
  name TEXT NOT NULL
);

COMMENT ON TABLE foo IS 'Windows Phone 8, however, was never able to overcome a long string of disappointments for Microsoft. ';
```

`sqlfluff lint` throws `L016 | Line is too long.` error for the above with the configurations mentioned below.

I'm not sure if this is working as expected. But shouldn't the `COMMON ON TABLE` statement be ignored ater specifying thr `ignore_comment_clauses` in `.sqlfluff`? I checkout out https://github.com/sqlfluff/sqlfluff/issues/2554 but it appears to address inline comments?

### Expected Behaviour

`COMMENT ON` clauses should be ignored after specifying `ignore_comment_clauses = True`

### Observed Behaviour

Throws a `L016 | Line is too long` error even after specifying `ignore_comment_clauses = True` in the config.

### How to reproduce

Use the above mentioned SQL command and run `sqlfluff lint` against it.

### Dialect

postgres

### Version

```
λ  ~ sqlfluff --version
sqlfluff, version 1.0.0
```

### Configuration

```python
[sqlfluff]
dialect = postgres
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all
exclude_rules = L048, L031

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 100
indent_unit = space
comma_style = trailing

# Keywords
[sqlfluff:rules:L010]
capitalisation_policy = upper

# Line length
[sqlfluff:rules:L016]
ignore_comment_lines = True
ignore_comment_clauses = True
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
IIRC, the code that ignores comment clauses is pretty specific about the syntax. We probably need to _either_ generalize the rule _or_ (better if appropriate) modify the dialect(s) so comment clauses can be recognized without requiring overly complex logic.
